plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1263,7 +1263,8 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
                     statisticsAccess.deleteExtendedStatistics(session, schemaTableName, location);
                 }
                 else {
-                    setRollback(() -> deleteRecursivelyIfExists(fileSystem, deltaLogDirectory));
+                    // deleteRecursivelyIfNothingExists ensures current CREATE TABLE doesn't delete directory if there's a conflict
```

Comment:
Why aren't we checking in the `catch` clause whether we're dealing with a `TransactionConflictException` instead?
By doing this, we'd likely know whether we're in a concurrency situation.

Reply:
@findinpath the rollback happens in a different thread AFAIK, the exception context needs to be passed on as well. Also the rollback initialisation happens in `beginCreateTable & createTable` calls which is much prior to `finishCreateTable` which is later.

Reply:
We're figuring out that another process already created the table only when trying to write the first transaction log file right?

Reply:
yes @findinpath, https://github.com/trinodb/trino/blob/5b82e10e632cb81e265d2331bbbfc00df8123b0b/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L1660
this is where the operation fails due Conflict.